CommonApi/src/main/java/mezz/jei/api/recipe/category/extensions/IGlobalRecipeCategoryExtension.java

```diff
@@ -0,0 +1,67 @@
+	 * this extension is registered to has drawn.
+	 * Tooltips are handled by {@link #getTooltipStrings(Object, IRecipeCategory, IRecipeSlotsView, double, double)} and
+	 * {@link #decorateExistingTooltips(List, Object, IRecipeCategory, IRecipeSlotsView, double, double)}.
+	 *
+	 * @param recipe          the current recipe being drawn.
+	 * @param recipeCategory  the recipe category of the recipe.
+	 * @param recipeSlotsView a view of the current recipe slots being drawn.
+	 * @param guiGraphics     the current {@link GuiGraphics} for rendering.
+	 * @param mouseX          the X position of the mouse, relative to the recipe.
+	 * @param mouseY          the Y position of the mouse, relative to the recipe.
+	 *
+	 * @see IDrawable for a simple class for drawing things.
+	 * @see IGuiHelper for useful functions.
+	 * @see IRecipeSlotsView for information about the ingredients that are currently being drawn.
+	 */
+	default void draw(T recipe, IRecipeCategory<T> recipeCategory, IRecipeSlotsView recipeSlotsView, GuiGraphics guiGraphics, double mouseX, double mouseY) {
+
+	}
+
+	/**
+	 * Get the tooltip for whatever is under the mouse.
+	 * These are additional tooltips to the ones added by the {@link IRecipeCategory} that
+	 * this extension is registered to.
+	 *
+	 * @param recipe          the current recipe being drawn.
+	 * @param recipeCategory  the recipe category of the recipe.
+	 * @param recipeSlotsView a view of the current recipe slots being drawn.
+	 * @param mouseX          the X position of the mouse, relative to the recipe.
+	 * @param mouseY          the Y position of the mouse, relative to the recipe.
+	 * @return tooltip strings. If there is no tooltip at this position, return an empty list.
+	 */
+	default List<Component> getTooltipStrings(T recipe, IRecipeCategory<T> recipeCategory, IRecipeSlotsView recipeSlotsView, double mouseX, double mouseY) {
+		return List.of();
+	}
+
+	/**
+	 * Allows modifying existing tooltips added by the {@link IRecipeCategory} that
+	 * this extension is registered to.
+	 * This is called before {@link #getTooltipStrings(Object, IRecipeCategory, IRecipeSlotsView, double, double)}
+	 * is called. This means the extension tooltips will not be included in the tooltip list.
+	 *
+	 * @param tooltips        the existing tooltip strings.
+	 * @param recipe          the current recipe being drawn.
+	 * @param recipeCategory  the recipe category of the recipe.
+	 * @param recipeSlotsView a view of the current recipe slots being drawn.
+	 * @param mouseX          the X position of the mouse, relative to the recipe.
+	 * @param mouseY          the Y position of the mouse, relative to the recipe.
+	 * @return tooltip strings. If there is no tooltip at this position, return an empty list.
+	 */
+	default List<Component> decorateExistingTooltips(List<Component> tooltips, T recipe, IRecipeCategory<T> recipeCategory, IRecipeSlotsView recipeSlotsView, double mouseX, double mouseY) {
```

Comment:
I do not see much utility in having this method *and* `getTooltipStrings`.
Could everything be done in this method, and we can remove `getTooltipStrings`?

Reply:
Sure but I thought it might be hard to notice the behavior when you have to pass the original list back. If some modders return their own list without the already existing elements of the category, they will essentially delete them.
I'll merge the two methods and will make sure to properly mention it in the doc.

Reply:
What is the use-case for altering the existing tooltip?
Should this interface be used to optionally override the old tooltip, without passing it in?

Reply:
You have three opportunities with the decorate method:
1. Remove all original tooltips of the category this extension is registered to by always returning an empty list or only with your custom tooltips.
2. Change existing tooltips of the category by modifying the elements in the list.
3. Add new custom tooltips by appending them to the list.

The method is able to basically do anything with the tooltips of the category it's bound to. This makes the `getTooltipStrings` method obsolete, that's why I decided to remove it.
Each of the things mentioned above has its own use cases. For example, staging mods could disable all tooltips until the player has the stage for the recipe. Or the staging mod draws an icon/text for the required stage on the category and adds a tooltip to it.